### PR TITLE
Feature/internal traps

### PR DESCRIPTION
**Pull request by louismunro**
For testing and inclusion in 4.4.

Adds the code changes to enable internal traps through the WebAPI. 
Adds the GUI setting to control the delay before VLAN reassignment or deassociation.

### CODE REVIEW

File: lib/pf/api.pm [hunk 1]
```diff
@@ -216,6 +218,9 @@ sub desAssociate {
 
     my ($switchdeauthMethod, $deauthTechniques) = $switch->deauthTechniques($switch->{'_deauthMethod'});
 
+    # sleep long enought to give the device enough time to fetch the redirection page.
```
Comment (julsemaan) [inline lib/pf/api.pm]: *sleep long enough

**Comment by fdurand** [comment]
In api.pm you forgot to add sleep for firewall sub. (inline)

**Comment by louismunro** [comment]
I did not forget.
Based on Derek's testing it seemed not to be necessary when n inline mode
snce the server is always reachable.

Testing will show whether that assumption is correct.

File: lib/pf/api.pm [hunk 2]
```diff
@@ -194,6 +194,8 @@ sub ReAssignVlan {
 
     my $switch = pf::SwitchFactory->getInstance()->instantiate( $postdata{'switch'} );
 
+    sleep $pf::config::Config{'vlan'}{'wait_for_redirect'};
```
Comment (julsemaan) [inline lib/pf/api.pm]: sleep $pf::config::Config{'trapping'}{'wait_for_redirect'};

It's in the trapping section

File: lib/pf/api.pm [hunk 3]
```diff
@@ -216,6 +218,9 @@ sub desAssociate {
 
     my ($switchdeauthMethod, $deauthTechniques) = $switch->deauthTechniques($switch->{'_deauthMethod'});
 
+    # sleep long enought to give the device enough time to fetch the redirection page.
+    sleep $pf::config::Config{'vlan'}{'wait_for_redirect'};
```
Comment (julsemaan) [inline lib/pf/api.pm]: same here

**Pull request merged**